src/commands/registries/azure/deployImageToAzure.ts

```diff
@@ -60,23 +66,16 @@ export async function deployImageToAzure(context: IActionContext, node?: RemoteT
     window.showInformationMessage(createdNewWebApp);
 }
 
-async function getNewSiteConfig(node: RemoteTagTreeItemBase): Promise<SiteConfig> {
+async function getNewSiteConfig(node: RemoteTagTreeItem): Promise<SiteConfig> {
     let registryTI: RegistryTreeItemBase = node.parent.parent;
 
     let username: string | undefined;
     let password: string | undefined;
-    let imagePath: string;
     let appSettings: NameValuePair[] = [];
     if (registryTI instanceof DockerHubNamespaceTreeItem) {
-        imagePath = registryTI.namespace;
         username = registryTI.parent.username;
-        password = registryTI.parent.password;
-
-        if (!username || !password) {
-            throw new Error("Failed to get credentials for Docker Hub.");
-        }
-    } else {
-        imagePath = registryTI.host;
+        password = await registryTI.parent.getPassword();
```

Comment:
There used to be an error thrown here if either username or password were undefined.  Does `getPassword` throw an error if it fails to retrieve the password from the keytar or whatever it's using?

Reply:
99% of the time there will be a username and password. It's not clear what should happen if there isn't, but I decided to err on the side of letting them deploy. There are plenty of self-hosted registries out there with weird auth schemes and I don't want to block people.